framework/common/error.h

```diff
@@ -81,15 +81,14 @@ class VulkanException : public std::runtime_error
 }        // namespace vkb
 
 /// @brief Helper macro to test the result of Vulkan calls which can return an error.
-#define VK_CHECK(x)                                                 \
```

Comment:
This change will affect a large portion of the samples, so I'm unsure as to why this change is required. We recently fixed batch mode and I didn't notice any problems in that regard.

Reply:
This turned out to be important for samples where pipeline construction can fail on macOS/iOS due to SPIR-V to MSL conversion errors caused by lack of support for certain SPIR-V language features, even with the extension showing as supported.  For instance, in the **fragment_shader_barycentric** sample, the Vulkan barycentrric extension is available on MoltenVK, but the SPIR-V language feature `PerVertexKHR` is not currently supported by MSL and `BaryCoordKHR` and `BaryCoordNoPerspKHR` cannot be activated at the same time.  Since SPIR-V to MSL compilation happens on the fly during pipeline construction, you need to check for failures here using `VK_CHECK` like this...

`VK_CHECK(vkCreateGraphicsPipelines(...))`

The previous VK_CHECK would fail out with a hard exit and cause batch mode to stop.  My suggested modification will trigger the error recovery code built into the framework to catch these, stop the current sample, and move on to the next.  Batch mode may work fine on other platforms that have full Vulkan support, but does not work well on portability implementations like MoltenVK where partial support is present not just at the extension level but at the SPIR-V feature level as well.  I did test this change with no observed negative effects on other platforms (Windows for now, linux coming later today).  I will continue to test and will report back if I find issues.

Reply:
I have just completed testing on linux (Manjaro) and the new VK_CHECK functionality seems to work fine on that platform.

Reply:
Agreed a throw here is better than an abort. This may cause some more crashes in batch mode from samples not correctly checking handles before deallocating. That is a good thing as it will highlight where samples can be improved to be more correct